### ISSUE TITLE
[Bug] mkiocccentry states that the IOCCC registration process and submit servers are not ready

### ISSUE THREAD

**Issue by lcn2** [ioccc-src/mkiocccentry#1037]
### Is there an existing issue for this?

- [X] I have searched for existing issues and did not find anything like this

### Describe the bug

At the end of pressing a submission, `mkiocccentry(1)` states:

```
The IOCCC registration process and submit servers are not ready.  You cannot
register for the IOCCC or use the IOCCC submit server at this time.
Please try again later, and monitor the IOCCC news for updates:
```

### What you expect

The `mkiocccentry(1)` tool would reference the FAQ in the "other repo" and perhaps the "next/register.html" and "next/submit.html" pages that are to be created (... wait for an commit in that "other repo").

### Environment

- OS: OS
- Device: Device
- Compiler: Compiler

### bug_report.sh output

This bug was reported by Eliot Lear.

It appears this will require an update from the previous code freeze.

### Anything else?

**SIGH** 😔

**Comment by lcn2** [comment]
See issue #1038

**Comment by xexyl** [comment]
I can do this and sync the fix in jstrdecode at the same time. 

Do you want me to do that? Also should the tool by default validate JSON?

Or I can not worry about this part.

**Comment by lcn2** [comment]
> I can do this and sync the fix in jstrdecode at the same time.
> 
> Do you want me to do that?

Regarding [GH-issuecomment-2561259252](https://github.com/ioccc-src/temp-test-ioccc/issues/5#issuecomment-2561259252), please open up a new issue to discuss this problem, @xexyl.  And yes, you can fix this now.

**Comment by lcn2** [comment]
> Also should the tool by default validate JSON?
> 
> Or I can not worry about this part.

Please discuss in issue #1038 , @xexyl

**Comment by lcn2** [comment]
We have a commit on the "other repo" that will change how the FAQ is used and that adds a new page.  Stay tuned.

**Comment by lcn2** [comment]
We believe we have addressed all of the current questions that still need answering at this time. If we've missed something or something else needs to be clarified, please ask again.

**Comment by xexyl** [comment]
> We have a commit on the "other repo" that will change how the FAQ is used and that adds a new page. Stay tuned.

Thanks .. let me know when it's done please.

**Comment by xexyl** [comment]
So: do you wish me to do this along with other bug fixes? I do have some questions but as those apply to other issues I'll save them for those issues.

**Comment by lcn2** [comment]
> > We have a commit on the "other repo" that will change how the FAQ is used and that adds a new page. Stay tuned.
> 
> Thanks .. let me know when it's done please.

It is done.

**Comment by lcn2** [comment]
> So: do you wish me to do this along with other bug fixes? I do have some questions but as those apply to other issues I'll save them for those issues.

Yes please.

**Comment by xexyl** [comment]
> > So: do you wish me to do this along with other bug fixes? I do have some questions but as those apply to other issues I'll save them for those issues.
> 
> Yes please.

Now I reread it I'm not sure if I'm clear what should be happening. Please clarify if this is it.

I believe that instead of printing out what it currently states it should print out a link to the FAQ on how to submit to the contest AND point to the register.html page too. But it should obviously be the `https://www.ioccc.org/next/register.html`. And it might be good to point out some other things too though right now i'm too tired even to elaborate on that.

I believe this is what you need. From the sound of it it is. I might go for it but I do at least have other fixes. However for now Ir really must go afk ... have at least three maybe four of these issues resolved I think.

**Comment by lcn2** [comment]
> > > So: do you wish me to do this along with other bug fixes? I do have some questions but as those apply to other issues I'll save them for those issues.
> 
> > 
> 
> > Yes please.
> 
> 
> 
> Now I reread it I'm not sure if I'm clear what should be happening. Please clarify if this is it.
> 
> 
> 
> I believe that instead of printing out what it currently states it should print out a link to the FAQ on how to submit to the contest AND point to the register.html page too. But it should obviously be the `https://www.ioccc.org/next/register.html`. And it might be good to point out some other things too though right now i'm too tired even to elaborate on that.
> 
> I believe this is what you need. From the sound of it it is. I might go for it but I do at least have other fixes. However for now Ir really must go afk ... have at least three maybe four of these issues resolved I think.

This seems like a reasonable approach, please proceed @xexyl

**Comment by xexyl** [comment]
> > > > So: do you wish me to do this along with other bug fixes? I do have some questions but as those apply to other issues I'll save them for those issues.
> > 
> > 
> > > 
> > 
> > 
> > > Yes please.
> > 
> > 
> > Now I reread it I'm not sure if I'm clear what should be happening. Please clarify if this is it.
> > I believe that instead of printing out what it currently states it should print out a link to the FAQ on how to submit to the contest AND point to the register.html page too. But it should obviously be the `https://www.ioccc.org/next/register.html`. And it might be good to point out some other things too though right now i'm too tired even to elaborate on that.
> > I believe this is what you need. From the sound of it it is. I might go for it but I do at least have other fixes. However for now Ir really must go afk ... have at least three maybe four of these issues resolved I think.
> 
> This seems like a reasonable approach, please proceed @xexyl

Is this post IOCCC28 or before?

**Comment by lcn2** [comment]
> > > > > So: do you wish me to do this along with other bug fixes? I do have some questions but as those apply to other issues I'll save them for those issues.
> 
> > > 
> 
> > > 
> 
> > > > 
> 
> > > 
> 
> > > 
> 
> > > > Yes please.
> 
> > > 
> 
> > > 
> 
> > > Now I reread it I'm not sure if I'm clear what should be happening. Please clarify if this is it.
> 
> > > I believe that instead of printing out what it currently states it should print out a link to the FAQ on how to submit to the contest AND point to the register.html page too. But it should obviously be the `https://www.ioccc.org/next/register.html`. And it might be good to point out some other things too though right now i'm too tired even to elaborate on that.
> 
> > > I believe this is what you need. From the sound of it it is. I might go for it but I do at least have other fixes. However for now Ir really must go afk ... have at least three maybe four of these issues resolved I think.
> 
> > 
> 
> > This seems like a reasonable approach, please proceed @xexyl
> 
> 
> 
> Is this post IOCCC28 or before?

Before. It would be part of the version 2.1 release of this repo.

**Comment by xexyl** [comment]
> > > > > > So: do you wish me to do this along with other bug fixes? I do have some questions but as those apply to other issues I'll save them for those issues.
> > 
> > 
> > > >
> > 
> > 
> > > >
> > 
> > 
> > > > >
> > 
> > 
> > > >
> > 
> > 
> > > >
> > 
> > 
> > > > > Yes please.
> > 
> > 
> > > >
> > 
> > 
> > > >
> > 
> > 
> > > > Now I reread it I'm not sure if I'm clear what should be happening. Please clarify if this is it.
> > 
> > 
> > > > I believe that instead of printing out what it currently states it should print out a link to the FAQ on how to submit to the contest AND point to the register.html page too. But it should obviously be the `https://www.ioccc.org/next/register.html`. And it might be good to point out some other things too though right now i'm too tired even to elaborate on that.
> > 
> > 
> > > > I believe this is what you need. From the sound of it it is. I might go for it but I do at least have other fixes. However for now Ir really must go afk ... have at least three maybe four of these issues resolved I think.
> > 
> > 
> > > 
> > 
> > 
> > > This seems like a reasonable approach, please proceed @xexyl
> > 
> > 
> > Is this post IOCCC28 or before?
> 
> Before. It would be part of the version 2.1 release of this repo.
> 
> ## UPDATE 0
> And see also [GH-issuecomment-2563061429](https://github.com/ioccc-src/mkiocccentry/issues/1038#issuecomment-2563061429)

Okay. So if I'm not mistaken .. tomorrow morning I have to do the following.

0. Run make www with the jstrdecode installed from jparse (If all okay then sync jparse to mkiocccentry).
1. Add the appropriate links rather than say the submit server is not ready.

With these done I think all the issues that are not post IOCCC28 could be marked complete?

As for it being before or post IOCCC28 I'm not sure what happened there. I know I wondered this about the script that checks for Makefile.local files but this one has to be done first. Anyway can you think of anything else for tomorrow with this repo?

About to leave for the day.

**Comment by lcn2** [comment]
> Okay. So if I'm not mistaken .. tomorrow morning I have to do the following.
> 
> Run make www with the jstrdecode installed from jparse (If all okay then sync jparse to mkiocccentry).
> Add the appropriate links rather than say the submit server is not ready.
> With these done I think all the issues that are not post IOCCC28 could be marked complete?

Correct.

> As for it being before or post IOCCC28 I'm not sure what happened there. I know I wondered this about the script that checks for Makefile.local files but this one has to be done first. Anyway can you think of anything else for tomorrow with this repo?

That seems to be a complete list.

> About to leave for the day.

Best wishes for having the rest of a good day @xexyl

**Comment by lcn2** [comment]
We believe we have addressed all of the current questions that still need answering at this time. If we've missed something or something else needs to be clarified, please ask again.

**Comment by xexyl** [comment]
> > Okay. So if I'm not mistaken .. tomorrow morning I have to do the following.
> > Run make www with the jstrdecode installed from jparse (If all okay then sync jparse to mkiocccentry).
> > Add the appropriate links rather than say the submit server is not ready.
> > With these done I think all the issues that are not post IOCCC28 could be marked complete?
> 
> Correct.

Okay that's good to know!

> 
> > As for it being before or post IOCCC28 I'm not sure what happened there. I know I wondered this about the script that checks for Makefile.local files but this one has to be done first. Anyway can you think of anything else for tomorrow with this repo?
> 
> That seems to be a complete list.

Great.

> 
> > About to leave for the day.
> 
> Best wishes for having the rest of a good day @xexyl

And you too! About to go get some food and then just take it easy for the last hours of my day (which is still quite a few but as you know I do go to bed earlier than most people).

**Comment by lcn2** [comment]
Please remember that completing tasks (which exclude all issues in this repo) need to resume the **Great Fork Merge** remain the **TOP PRIORITY**.

It is our goal to finish the Great Fork Merge by or on Saturday 2024 Dec 28.

**Comment by xexyl** [comment]
I have completed this task but I have to go afk for a bit ... I'll also include the updates of the temp-test-ioccc to the IOCCC website. Unless there is something else I am missing I believe that will mean that the 2.1 release can happen!

However I do have to go afk .. though I might try slipping in this first part first .. not sure.

**Comment by xexyl** [comment]
Okay I did the other part too. However before I do a pull request I have to be afk a while ... when back I'll do this and then once you've done the great fork merge the URLs should be okay (though we should of course check them)!

**Comment by xexyl** [comment]
.. actually I'll quickly do it if all is okay with `make release`. Then afk a while. But since that's probably all I need to do today, except maybe the list of words ... that's okay.

**Comment by lcn2** [comment]
# Background

This issue will have to be resolved in release 2.2 of this repo as the **Great Fork Merge** had to happen last night and we needed to move forward.

# TOP PRIORITY

This needs to be resolved in early January as people looking into submitting to the IOCCC are looking at this repo NOW. We need to push out a version 2.2 of those repo that includes resolving this issue in early January,

This is why this is a top priority.

**Comment by xexyl** [comment]
Well I left a comment over in the last pull request asking for clarification on something. If you can reply that would be helpful. 

But it could also be that if you don't get to it I can copy it over here. 

Of course the next step is adding a function in the lower level code of chkentry for that tool and the other two as well. 

That new function should not take long to write but it might not be until tomorrow morning. I have to do other things now. 

Hopefully with clarification on this issue I can take care of it too.

**Comment by lcn2** [comment]
> Well I left a comment over in the last pull request asking for clarification on something. If you can reply that would be helpful.
> 
> But it could also be that if you don't get to it I can copy it over here.
> 
> 

We looked but did't find what you are referring to, @xexyl, sorry (tm Canada 🇨🇦).  There is a "whirlwind of things going on" as the expression goes and so we may have overlooked it by mistake.

Please let a link to the comment here, or copy it over as you prefer.

**Comment by xexyl** [comment]
> > Well I left a comment over in the last pull request asking for clarification on something. If you can reply that would be helpful.
> > 
> > But it could also be that if you don't get to it I can copy it over here.
> > 
> > 
> 
> We looked but did't find what you are referring to, @xexyl, sorry (tm Canada 🇨🇦).  There is a "whirlwind of things going on" as the expression goes and so we may have overlooked it by mistake.
> 
> Please let a link to the comment here, or copy it over as you prefer.

I am not surprised you missed it: it was kind of buried deep above (how does that work ?) several additional comments. 

This text is what I meant:

> This new text. Is that when the submit server is ready (that macro is defined) or when it's not (macro undefined)? Or do you want the macro to no longer be there?

The text that refers to is the way you wanted me to change the current text in that function. 

Once you clarify that I can do it although I have the other things I think should be done first.

**Comment by lcn2** [comment]
> > > Well I left a comment over in the last pull request asking for clarification on something. If you can reply that would be helpful.
> > > But it could also be that if you don't get to it I can copy it over here.
> > 
> > 
> > We looked but did't find what you are referring to, @xexyl, sorry (tm Canada 🇨🇦).  There is a "whirlwind of things going on" as the expression goes and so we may have overlooked it by mistake.
> > Please let a link to the comment here, or copy it over as you prefer.
> 
> I am not surprised you missed it: it was kind of buried deep above (how does that work ?) several additional comments.
> 
> This text is what I meant:
> 
> > This new text. Is that when the submit server is ready (that macro is defined) or when it's not (macro undefined)? Or do you want the macro to no longer be there?
> 
> The text that refers to is the way you wanted me to change the current text in that function.
> 
> Once you clarify that I can do it although I have the other things I think should be done first.

Does [GH-issuecomment-2565849662](https://github.com/ioccc-src/mkiocccentry/issues/723#issuecomment-2565849662) clarify your question, @xexyl ?

**Comment by xexyl** [comment]
> > > > Well I left a comment over in the last pull request asking for clarification on something. If you can reply that would be helpful.
> > > > But it could also be that if you don't get to it I can copy it over here.
> > > 
> > > 
> > > We looked but did't find what you are referring to, @xexyl, sorry (tm Canada 🇨🇦).  There is a "whirlwind of things going on" as the expression goes and so we may have overlooked it by mistake.
> > > Please let a link to the comment here, or copy it over as you prefer.
> > 
> > 
> > I am not surprised you missed it: it was kind of buried deep above (how does that work ?) several additional comments.
> > This text is what I meant:
> > > This new text. Is that when the submit server is ready (that macro is defined) or when it's not (macro undefined)? Or do you want the macro to no longer be there?
> > 
> > 
> > The text that refers to is the way you wanted me to change the current text in that function.
> > Once you clarify that I can do it although I have the other things I think should be done first.
> 
> Does [GH-issuecomment-2565849662](https://github.com/ioccc-src/mkiocccentry/issues/723#issuecomment-2565849662) clarify your question, @xexyl ?

No. Two different things. I'm talking about this issue. You thought the text could be improved in the function that shows the submit server URL. See my comment in that pull request: [discussion_r1899184755](https://github.com/ioccc-src/mkiocccentry/pull/1042#discussion_r1899184755).

I don't know exactly where you want the text you proposed in the comment above that one. Maybe it'd be good to try rewording it and then I can hopefully take care of it so that the issue can be closed as complete.

**Comment by lcn2** [comment]
> No. Two different things. I'm talking about this issue. You thought the text could be improved in the function that shows the submit server URL. See my comment in that pull request: discussion_r1899184755.
> 
> I don't know exactly where you want the text you proposed in the comment above that one. Maybe it'd be good to try rewording it and then I can hopefully take care of it so that the issue can be closed as complete.

Instead of the text stating:

```
The IOCCC registration process and submit servers are not ready.  You cannot
register for the IOCCC or use the IOCCC submit server at this time.
Please try again later, and monitor the IOCCC news for updates:

Some URL
```

The modified text as proposed higher up could end with:

```
... ((the rest of the new text that referred to uploading the filename into the slot)) ... to the IOCCC submit server:

    https://submit.ioccc.org

When the contest is status is open.  For the contest status see:

    https://www.ioccc.org/status.html
```

We cannot locate the original proposed replacement text right now, but hopefully that should give you an idea.

Does that help?

**Comment by xexyl** [comment]
> > No. Two different things. I'm talking about this issue. You thought the text could be improved in the function that shows the submit server URL. See my comment in that pull request: discussion_r1899184755.
> > I don't know exactly where you want the text you proposed in the comment above that one. Maybe it'd be good to try rewording it and then I can hopefully take care of it so that the issue can be closed as complete.
> 
> Instead of the text stating:
> 
> ```
> The IOCCC registration process and submit servers are not ready.  You cannot
> register for the IOCCC or use the IOCCC submit server at this time.
> Please try again later, and monitor the IOCCC news for updates:
> 
> Some URL
> ```
> 
> The modified text as proposed higher up could end with:
> 
> ```
> ... ((the rest of the new text that referred to uploading the filename into the slot)) ... to the IOCCC submit server:
> 
>     https://submit.ioccc.org
> 
> When the contest is status is open.  For the contest status see:
> 
>     https://www.ioccc.org/status.html
> ```
> 
> We cannot locate the original proposed replacement text right now, but hopefully that should give you an idea.
> 
> Does that help?

The original was [here](https://github.com/ioccc-src/mkiocccentry/pull/1042#discussion_r1899024879).

The text is not a problem. The problem is when should it be shown. In particular there is currently a macro that is defined to say the server is ready. But when it's undefined it has other text. Right now it says, because the submit server is ready (unless of course the id used was 'test'):

```c
   /*
     * print the submit server URL
     */
    para("",
         "If the contest is still open, you may upload the above",
         "tarball to the following submission URL, assuming you've",
         "registered and changed your initial password:",
         "",
         NULL);
```

But I noticed something odd now.

In that same function is a macro that is:

```c
#elif defined (IOCCC_REGISTRATION_READY)

    /*
     * print that the submit server not ready for submissions
     */
    para("",
         "The IOCCC submit server is NOT ready to receive submissions.",
         "Please try again later, and monitor the IOCCC news for updates:",
         "",
         NULL);
```

I would think that might be an else condition. There is a function to show the registration URL too.

Perhaps these two functions need to be rewritten entirely? If yes the question is how do you want the macros to be dealt with? As in do both functions need macros and have different text depending on whether it's defined or not?

If it should depend on the text what should it say for both ready and not ready (the comment I linked to above hopefully helps there).

The other function is `show_submit_url()` and I think I see the problem now. There are macros that are not defined and that do not even exist in the soup/limit_ioccc.h file.

So I think what might have to happen (please correct me if I'm wrong) is:

0. The first function to be called is the registration URL one.
1. The second function to be called is the submit server URL.

Then in the registration URL function it will say that if the contest is pending or open one may register at ... (URL of registration URL). It is not clear to me if we need a macro or not. Please let me know on that.

Then as for the submit function it should say that if it's open status and you have already registered you can then upload your submission. It would tell you the filename and slot number as well. If the submit server is NOT ready it would print something else but what I don't know.

What do you think? I'll look at this tomorrow morning but hopefully this clarifies things better. Just let me know and we'll go from there.

**Comment by lcn2** [comment]
We think that the `IOCCC_REGISTRATION_READY` define is no longer needed.

This need when we didn't have a submit server, nor did we know what hostname to would have.

The code for when `IOCCC_REGISTRATION_READY` is **NOT** defined is obsolete / no longer needed.  The define would be removed.

**Comment by lcn2** [comment]
> The original was [here](https://github.com/ioccc-src/mkiocccentry/pull/1042#discussion_r1899024879).

Thanks.  So based on "the original" and [GH-issuecomment-2566012614](https://github.com/ioccc-src/mkiocccentry/issues/1037#issuecomment-2566012614) we recommend:

```
When the contest is open (see https://www.ioccc.org/submit.html), you must upload
((put filename here)) into slot ((put entry number here)) on the IOCCC submit server:

  https://submit.ioccc.org

For the contest status see:

    https://www.ioccc.org/status.html

For more information, see the FAQ 0.0 at:

  https://www.ioccc.org/faq.html#enter
```

> The text is not a problem. The problem is when should it be shown. In particular there is currently a macro that is defined to say the server is ready. But when it's undefined it has other text. Right now it says, because the submit server is ready (unless of course the id used was 'test'):
> 
> ```c
>    /*
>      * print the submit server URL
>      */
>     para("",
>          "If the contest is still open, you may upload the above",
>          "tarball to the following submission URL, assuming you've",
>          "registered and changed your initial password:",
>          "",
>          NULL);
> ```
> 
> But I noticed something odd now.
> 
> In that same function is a macro that is:
> 
> ```c
> #elif defined (IOCCC_REGISTRATION_READY)
> 
>     /*
>      * print that the submit server not ready for submissions
>      */
>     para("",
>          "The IOCCC submit server is NOT ready to receive submissions.",
>          "Please try again later, and monitor the IOCCC news for updates:",
>          "",
>          NULL);
> ```
> 
> I would think that might be an else condition. There is a function to show the registration URL too.

See [GH-issuecomment-2566119534](https://github.com/ioccc-src/mkiocccentry/issues/1037#issuecomment-2566119534).  TL;DR the `IOCCC_REGISTRATION_READY` define is no longer needed.

> Perhaps these two functions need to be rewritten entirely? If yes the question is how do you want the macros to be dealt with? As in do both functions need macros and have different text depending on whether it's defined or not?
> 
> If it should depend on the text what should it say for both ready and not ready (the comment I linked to above hopefully helps there).

Well the function that displays stuff could be re-written.  The submit server is ready and will remain ready even the contest status is not open.

> The other function is `show_submit_url()` and I think I see the problem now. There are macros that are not defined and that do not even exist in the soup/limit_ioccc.h file.
> 
> So I think what might have to happen (please correct me if I'm wrong) is:
> 
> 0. The first function to be called is the registration URL one.
> 1. The second function to be called is the submit server URL.
> 
> Then in the registration URL function it will say that if the contest is pending or open one may register at ... (URL of registration URL). It is not clear to me if we need a macro or not. Please let me know on that.
> 
> Then as for the submit function it should say that if it's open status and you have already registered you can then upload your submission. It would tell you the filename and slot number as well. If the submit server is NOT ready it would print something else but what I don't know.
> 
> What do you think? I'll look at this tomorrow morning but hopefully this clarifies things better. Just let me know and we'll go from there.

Your idea seems reasonable, give that the submit server is always ready.

BTW: Just because the submit server is always ready does **NOT** mean one can login, nor that one can upload files, etc.

**Comment by xexyl** [comment]
Going to look at this next .. will start out replying but hopefully I won't need any more replies.

**Comment by xexyl** [comment]
> We think that the `IOCCC_REGISTRATION_READY` define is no longer needed.
> 
> This need when we didn't have a submit server, nor did we know what hostname to would have.
> 
> The code for when `IOCCC_REGISTRATION_READY` is **NOT** defined is obsolete / no longer needed. The define would be removed.

Hmm ... okay the question then is what about the submit server macro? My guess is it's needed because when the contest is closed it's not going to be 'ready' - as the guidelines/rules/FAQ say. I'll keep that in mind.

**Comment by xexyl** [comment]
Oh .. I see the submit server is also always ready. Okay so that macro is not needed either! The question is what to change the message to and whether or not we need both functions. Let me play with this one ...

**Comment by xexyl** [comment]
I think I'm about done with this .. just running `make all test`. I have to go momentarily but hoping I can get the commit in before I leave.

**Comment by xexyl** [comment]
Running make release .. then will do a make test (to see output) and if all okay I'll commit. Not sure if that'll be a bit later on or if I'll do it now but I hope to in a few minutes. If so I think this issue is probably resolved and the other one is by default also resolved. Then there will be no more top priority issues and we might be able to do a code freeze!

**Comment by lcn2** [comment]
Has this bug been resolved, @xexyl?

He think so but want to be sure before we declare repo version 2.3.

**Comment by xexyl** [comment]
> Has this bug been resolved, @xexyl?
> 
> He think so but want to be sure before we declare repo version 2.3.

I think it has: minus the slight format change you want. I'm working on that now .. just making sure it looks okay. Then if it seems fine to you (after commit) I would say yes go ahead with it.

**Comment by lcn2** [comment]
> > Has this bug been resolved, @xexyl?
> 
> > 
> 
> > He think so but want to be sure before we declare repo version 2.3.
> 
> 
> 
> I think it has: minus the slight format change you want. I'm working on that now .. just making sure it looks okay. Then if it seems fine to you (after commit) I would say yes go ahead with it.

Thanks @xexyl 

Correction to our various recent comments on 2024 Dec 31:

## UPDATE 0

Correction: repo version 2.3

The new version of this repo (with the suggested mod to `mkiocccentry(1)`) should be version 2.3 as we are already in repo version 2.2.1 according to the `CHANGES.md` file.

**Comment by xexyl** [comment]
I believe this issue is done now? It should also (I think) complete the other top priority issue?

**Comment by lcn2** [comment]
> I believe this issue is done now? It should also (I think) complete the other top priority issue?

Correct and thank you @xexyl
